akka-http/src/main/scala/akka/http/javadsl/server/directives/RouteDirectives.scala

```diff
@@ -112,11 +113,18 @@ abstract class RouteDirectives extends RespondWithDirectives {
   /**
    * Completes the request using the given status code, headers, and response entity.
    */
-  def complete(status: StatusCode, headers: java.lang.Iterable[HttpHeader], entity: RequestEntity) = RouteAdapter {
+  def complete(status: StatusCode, headers: java.lang.Iterable[HttpHeader], entity: ResponseEntity) = RouteAdapter {
     D.complete(scaladsl.model.HttpResponse(status = status.asScala, entity = entity.asScala, headers = Util.immutableSeq(headers).map(_.asScala))) // TODO avoid the map()
   }
 
   /**
+   * Completes the request using the given status code, headers, and response entity.
+   */
+  @deprecated("This directive is for binary compatibility only", "10.0.6")
```

Comment:
I think we should try out in which cases this overload is going to be used. If the most specific overload is picked, then it might be always this one (apart from `CloseDelimited` which isn't a `RequestEntity`). What we don't want is that every legitimate usage of `complete` is flagged with a deprecation warning.